packages/mongodb-chatbot-ui/src/useConversation.tsx

```diff
@@ -283,6 +300,36 @@ function conversationReducer(
         ),
       };
     }
+    case "appendStreamingReferences": {
+      if (!state.conversationId) {
+        console.error(
+          `Cannot appendStreamingResponse without a conversationId`
+        );
+        return state;
+      }
+      const { streamingMessage, streamingMessageIndex } = getStreamingMessage();
+      if (!streamingMessage) {
+        console.error(
+          `Cannot appendStreamingResponse without a streamingMessage. Make sure to dispatch createStreamingResponse first.`
+        );
+        return state;
+      }
+      const modifiedMessage = {
+        ...streamingMessage,
+        references: [
+          ...(streamingMessage.references ?? []),
+          ...action.data.map((ref) => addReferenceLinkVariant(ref)),
```

Comment:
given that this implementation is mongodb-specific but the UI lib is generalizable, can we make this function passed as a prop so that it's applicable to our app but other package consumers don't need to use?

(even if no-one else ever does that, i think it's still best practice to dependency inject something so application-specific given the rest of our patterns in the app)

Reply:
Moved to be a callback prop on `<Chatbot />`